File: home-server-ng/src/main/java/maroroma/homeserverng/iot/model/MiniSprite.java
package maroroma.homeserverng.iot.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * modele pour la description d'un sprite, avec l'état de ses pixels
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class MiniSprite {
    private String name;
    private String description;
    private List<List<MiniPixel>> lines;
}

File: home-server-ng/src/frontend/home-server-tsx/src/model/iot/MiniSprite.ts
import MiniPixel from "./MiniPixel";

export default class MiniSprite {

    static empty(): MiniSprite {

        const defaultGrid: MiniPixel[][] = [];


        for (let line = 0; line < 8; line++) {
            const newLine: MiniPixel[] = [];
            for (let column = 0; column < 8; column++) {
                newLine.push(new MiniPixel(false));
            }
            defaultGrid.push(newLine);
        }
        return new MiniSprite("", "", defaultGrid);
    }


    constructor(public name: string, public description: string, public lines: MiniPixel[][]) { }
}